### ISSUE TITLE
Document::addData() should be nullable

### ISSUE THREAD

**Issue by rgazelot** [tobscure/json-api#130]
I think that the method [Document::setData($element)](https://github.com/tobscure/json-api/blob/master/src/Document.php#L139) should accept a nullable value as the constructor does.

**Comment by a-komarev** [comment]
This will require to change signature to `public function setData(?ElementInterface $element)` what will lead drop PHP7.0 support. Because nullable types are available since PHP7.1 only.

**Comment by rgazelot** [comment]
What about `public function setData(ElementInterface $element = null)`?

**Comment by a-komarev** [comment]
By this way you will only be able to do `$document->setData();` but not `$docuemnt->setData(null)`. Try it yourself.

**Comment by rgazelot** [comment]
I agree that PHP7.1 nullable argument are by far the best way to do it, but I'm ok with the PHP7.0 compatibility. Maybe just keep in mind this thing for a future PHP7.1 support?

**Comment by franzliedke** [comment]
For PHP 7.0 compatibility, we would probably need a `clearData()` function then.

**Comment by f3ath** [comment]
What about creating a special case: `function setNullData(): void`?

**Comment by a-komarev** [comment]
Before implement any of proposed ways we need to define concrete use case for it.
